OutOfSchool/OutOfSchool.WebApi/Controllers/V2/WorkshopController.cs

```diff
@@ -146,18 +161,46 @@ public async Task<IActionResult> GetByFilter([FromQuery] WorkshopFilter filter,
     [Consumes("multipart/form-data")]
     public async Task<IActionResult> Create([FromForm] WorkshopV2Dto dto)
     {
-        var userHasRights = await IsUserProvidersOwner(dto.ProviderId).ConfigureAwait(false);
+        if (dto == null)
+        {
+            return BadRequest("Workshop is null.");
+        }
+
+        if (await IsProviderBlocked(dto.ProviderId).ConfigureAwait(false))
+        {
+            return StatusCode(403, "Forbidden to create workshops at blocked providers");
+        }
+
+        if (await IsCurrentUserBlocked())
+        {
+            return StatusCode(403, "Forbidden to create the workshop by the blocked provider.");
+        }
+
+        if (!ModelState.IsValid)
+        {
+            return BadRequest(ModelState);
+        }
+
+        var userHasRights = await this.IsUserProvidersOwnerOrAdmin(dto.ProviderId).ConfigureAwait(false);
         if (!userHasRights)
         {
             return StatusCode(403, "Forbidden to create workshops for another providers.");
         }
 
-        dto.Id = default;
-        dto.Address.Id = default;
+        try
+        {
+            var creationResult = await combinedWorkshopService.Create(dto).ConfigureAwait(false);
 
-        var creationResult = await combinedWorkshopService.Create(dto).ConfigureAwait(false);
+            // here we will get "false" if workshop was created by assistant provider admin
+            // because user is not currently associated with new workshop
+            // so we can update information to allow assistant manage created workshop
+            if (!await IsUserProvidersOwnerOrAdmin(creationResult.Workshop.ProviderId, creationResult.Workshop.Id).ConfigureAwait(false))
```

Comment:
it does not look like a controller logic, as for me it is more business logic and should be moved to the service